py-polars/polars/internals/frame.py

```diff
@@ -516,22 +536,28 @@ def read_parquet(
             Path to a file or a file like object. Any valid filepath can be used.
         columns
             Columns to select. Accepts a list of column indices (starting at zero) or a list of column names.
-        projection
-            Indices of columns to select. Note that column indices start at zero.
         n_rows
             Stop reading from parquet file after reading ``n_rows``.
         """
+        projection: Optional[tp.List[int]] = None
+        if columns:
+            if isinstance(columns, list):
+                if all(isinstance(i, int) for i in columns):
+                    projection = columns  # type: ignore
```

Comment:
@zundertj Do you know a way to tell mypy that columns canl only be `List[int]` at this point?

Reply:
Apologies, had been working on other projects. I have put a PR.